nhais/sequence/interchange_id.py

```diff
@@ -0,0 +1,15 @@
+from sequence.sequence_factory import get_sequence_generator
+
+
+class InterchangeIdGenerator:
```

Comment:
(Idea) This class has the same implementation as `MessageIdGenerator` but with different key value - maybe we should create generic generator and make key as input for it? I mean that we could have i.e. something like this:
````
interchange_key = InterchangeKey(sender, recipient)
interchange_id = IdGenerator().next(key)
````
That way generator code is extracted from key that's changing and we could easily reuse generators in different adapters

Reply:
@mattd-kainos , what do you think about it?

Reply:
It might be over-engineering after all... If this works and has just few lines than I probably stretched that too much

Reply:
Could do it that way, but I don't think it affects reusability because the reusable bit SequenceGenerator is already its own interface / implementation.

Another option would be to have a single class with methods to generate each type of identifier. The reasoning would be in case we need to add transactions in the future.

Reply:
A single class would also let us define the table_name in one place? The name of the dynamo table probably needs to be changed as well since it is now used for all sequence numbers.

Reply:
I think it is designed that way to store all ids in one table. Do we want each id to have it's own table?

Reply:
all ids use the same table